bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientAPIHandler.java

```diff
@@ -0,0 +1,401 @@
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.airgradient.internal.config.AirGradientAPIConfiguration;
+import org.openhab.binding.airgradient.internal.discovery.AirGradientLocationDiscoveryService;
+import org.openhab.binding.airgradient.internal.model.LedMode;
+import org.openhab.binding.airgradient.internal.model.Measure;
+import org.openhab.binding.airgradient.internal.prometheus.PrometheusMetric;
+import org.openhab.binding.airgradient.internal.prometheus.PrometheusTextParser;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link AirGradientAPIHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class AirGradientAPIHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirGradientAPIHandler.class);
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private final HttpClient httpClient;
+    private final Gson gson;
+
+    public AirGradientAPIHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = new Gson();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.info("Handle command {} for channel {}!", command.toFullString(), channelUID);
+        if (command instanceof RefreshType) {
+            pollingCode();
+        } else {
+            // This is read only
+            logger.debug("Received command {} for channel {}, but the API is read only", command.toString(),
+                    channelUID.getId());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        AirGradientAPIConfiguration config = getConfiguration();
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
```

Comment:
Maybe it is better to remove this executor.execute block and make it part of the `pollingCode` method. Because now this is fired, but if it takes some more time the `pollingCode()` is called in parallel. So you might and up with some timing issues. 
And `pollingcode` is also responsible for reporting online/offline anyway, so there is allready some duplicate code.

Reply:
Thanks! I removed polling completely, as we can detect communication issues when we need to communicate instead.